### PR TITLE
Refine inference on python

### PR DESCRIPTION
**Pull request by jerrywgz**
1. Automatically export infer_cfg.yml with the same path of inference model in export_model.py
2. Make cpp_infer.py independent of PaddleDetection
2. Support custom label_list as input
3. Support dump box to save inference result 
4. Replace tools/cpp_demo.yml by demo/infer_cfg.yml which is generated by export_model.py

### CODE REVIEW

File: tools/export_model.py [hunk 1]
```diff
@@ -30,6 +30,62 @@
 logger = logging.getLogger(__name__)
 
 
+def parse_cpp_config(config):
```
Comment (qingqing01) [inline tools/export_model.py]: dumpy_infer_config

File: tools/export_model.py [hunk 2]
```diff
@@ -30,6 +30,62 @@
...
+    cfg_name = os.path.basename(FLAGS.config).split('.')[0]
+    save_dir = os.path.join(FLAGS.output_dir, cfg_name)
+    cpp_cfg = {
```
Comment (qingqing01) [inline tools/export_model.py]: cpp_cfg -> cfg

cpp并不是一个很好的名字

File: tools/export_model.py [hunk 3]
```diff
@@ -30,6 +30,62 @@
...
+        'use_python_inference': False,
+        'mode': 'fluid',
+        'draw_threshold': 0.5,
+        'metric': config['metric']
+    }
+    cpp_arch_list = {'YOLO': 3, 'SSD': 40, 'RCNN': 40, 'RetinaNet': 40}
```
Comment (qingqing01) [inline tools/export_model.py]: trt_min_subgraph

Comment (qingqing01) [inline tools/export_model.py]: cpp_arch_list  -> trt_min_subgraph

Comment (qingqing01) [inline tools/export_model.py]: cpp_arch_list  -> trt_min_subgraph

File: tools/export_model.py [hunk 4]
```diff
@@ -30,6 +30,62 @@
...
+    arch = config['architecture']
+
+    for cpp_arch, min_subgraph_size in cpp_arch_list.items():
```
Comment (qingqing01) [inline tools/export_model.py]: cpp_arch -> arch

**Comment by jerrywgz** [comment]
> Could paste a dumped yml here

add a demo config in demo/infer_cfg.yml

File: tools/export_model.py (hunk 2 above)
Comment (jerrywgz) [reply tools/export_model.py]: Done

File: tools/export_model.py (hunk 3 above)
Comment (jerrywgz) [reply tools/export_model.py]: Done

Comment (jerrywgz) [reply tools/export_model.py]: Done

Comment (jerrywgz) [reply tools/export_model.py]: Done

File: tools/export_model.py (hunk 4 above)
Comment (jerrywgz) [reply tools/export_model.py]: Done

File: tools/export_model.py (hunk 1 above)
Comment (jerrywgz) [reply tools/export_model.py]: Done

**Pull request merged**